core/src/main/java/org/hibernate/ogm/remote/impl/DatabaseIdentifier.java

```diff
@@ -4,7 +4,7 @@
  * License: GNU Lesser General Public License (LGPL), version 2.1 or later
  * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
  */
-package org.hibernate.ogm.datastore.couchdb.util.impl;
```

Comment:
It doesn't have to be shared across all dialect, but it seems to have some methods the could be shared across some dialects.

I think I will avoid this change for now and create different classes. We can eventually share the code later as a different pull request